```python
if(condition):
    statement1
    statement2
    ...
    statementn
elif(condition2):
    statement31
    statement32
else:
    statement22
    statement23
    statement24
statement11
statement12
```

In [5]:
a = int(input())
b = int(input())

if(a % 2 == 0):
    if(b % 2 == 0):
        print('zoj')
    else:
        print('fard')
else:
    if(b % 2 == 1):
        print("zoj")
    else:
        print('fard')


10
2
zoj


In [ ]:
a = int(input())
b = int(input())

if((a+b) % 2 == 0):
    c = int(input())
    print(b*c)
else:
    print(a*b)

In [1]:
print(10)

10


## Nested If

```python
        
if(condition1):
    if(condition2):
        if(condition3):
            print(1)
        else:
            print(2)
    else:
        print(3)
else:
    print(4)
```

> Practice

## Loop

> Practice

In [10]:
n = int(input())

a = n%10
n = n//10
b = n%10
n = n//10
c = n%10
n = n//10
d = n%10
print(a, b, c, d)
print('sum: ', a+b+c+d)

2453
3 5 4 2
sum:  14


In [5]:
a = int(input('Enter a number: '))

cnt = 0
suma = 0
while(a != 0):
    #print(a)
    cnt = cnt + 1
    suma = suma + (a%10)
    #print(a % 10)
    a = a // 10

print(cnt, suma)

Enter a number: 200091110
9 14


In [1]:
m = int(input())
for i in range(4):
    a = int(input())
    if(a > m):
        m = a
    print('m: ',m)
        
print('max: ', m)

-8
-10
m:  -8
-4
m:  -4
87
m:  87
0
m:  87
max:  87


In [ ]:
m1 = int(input())
m2 = int(input())
if(m2 > m1):
    m1, m2 = m2, m1

for i in range(8):
    a = int(input())
    if(a >= m1):
        m2 = m1
        m1 = a
    elif(a >= m2):
        m2 = a
    print('m1: ',m1, 'm2: ', m2)
        
print('max1: ', m1, 'max2: ', m2)

-4
54
10
m1:  54 m2:  10
1000
m1:  1000 m2:  54
80
m1:  1000 m2:  80
0
m1:  1000 m2:  80
2000
m1:  2000 m2:  1000
9
m1:  2000 m2:  1000
491
m1:  2000 m2:  1000


In [5]:
a = int()
if(-5 > a):
    print(10)
else:
    print(20)

20


## Nested Loop

```python
for i in l:
    for j in m:
        print(10)
```

```****
****
****
****```

> Practice


```python
break, continue, pass
```


In [7]:
for i in range(5):
    print(i)
    if(i == 2):
        continue
        

0
*
1
*
2
3
*
4
*


###### float problem

In [11]:
a = float(32.3000)
print(a)

32.3


> answer?

##### new if

```python
x < y ? -1 : (x == y ? 0 : 1)

expression_if_true if condition else expression_if_false
```

In [14]:
print(1 if 10 < 2 else -1)

if(10 > 2):
    print(1)
else:
    print(-1)

-1
1


In [15]:
print(9.322e-36j)
9.322 + 10**-36 j


9.322e-36j


### Numbers
int | float | complex
:---: | :---: | :---: 
10 | 0.0 | 3.14j
100 | 15.20 | 45.j
-14 | -21.9	| 9.322e-36j
080	| 32.3+e18	 | .876j
-0x260 | -32.54e100	| 3e+26J


#### Number Type Conversion
```python
a = int(x)

a = float(x)

a = complex(x)

a = complex(x, y)

```

### Mathematical Functions


```python

abs(x)

math.ceil(x)

math.exp(x)```

> e**x

```python

math.floor(x)

math.log(x)```

> The natural log of x ( x > 0 )

```python

math.log10(x)

max(a, b, ...)

min(a, b, ...)

math.modf(x)

```

> The fractional and integer parts of x in a two-item tuple. Both parts have the same sign as x. The integer part is returned as a float.

```python

pow(x, y)

math.pow(x, y)

round(x[,n])```

> round( 1.5 ) = 1.0
>
> round( -1.5 ) = -2.0
>
> round( 1.178, 2 ) = 1.18

```python

math.sqrt(x)

```

> x > 0



### Random Number Functions

```python
random.choice(seq)
```
> random.choice ( [10, 2, 4, -10, 'salam'] )
> 
> random.choice ( 'hello' )

```python
random.randrange ([start,] stop [,step])

random.random()
```
> a random float number r. 0 <= r <= 1

```pyhton
random.seed([x])

random.shuffle(l)
```
> in place randomize

```python
random.uniform(x, y)
```
> a random float number r. x <= r <= y

### Trigonometric Functions

```python
math.acos(x)
```
> Return the arc cosine of x, in radians.

```python

math.asin(x)

math.atan(x)

math.atan2(x, y)

```
> Return atan(y / x), in radians.

```python
math.cos(x)
```

> Return the cosine of x radians.

```python
math.sin(x)

math.tan(x)

math.degrees(x)
```

> Converts angle x from radians to degrees.

```python
math.radians(x)
```
> Converts angle x from degrees to radians.



### Mathematical Constants

```python
print(math.pi)

print(math.e)
```

In [30]:
a = 10
print("%.2f" % (15000.2))

15000.20


### String Formatting Operator

```python
print( "My name is %s and weight is %d kg!" % ('Zara', 21) )
```
> My name is Zara and weight is 21 kg!

| Format Symbol | Conversion |
| :---: | :---: |
| %c | character |
| %s | string | 
| %d | signed decimal integer | 
| %o | octal integer |
| %x | hexadecimal integer (lowercase letters) |
| %X | hexadecimal integer (UPPERcase letters) |
| %e | exponential notation (with lowercase 'e') |
| %E | exponential notation (with UPPERcase 'E') |
| %f | floating point real number |




### Built-in String Methods

```python
a = 'salam'
a.capitalize()

a.center(width, fillchar)

a.count(str, beg=0, end=len(a))

b = a.find(str, beg=0, end=len(a))
```
> b = -1 or index

```python
a.index(str, beg=0, end=len(a))
```
> b = index or raise an exception

```python
a.isalnum()
```
> Returns true if string has at least 1 character and all characters are alphanumeric and false otherwise.

```python
a.isalpha()

a.isdigit()

a.islower()

a.isupper()

a.isnumeric()

a.join('*//*')
```
> print('salam'.join('-//-'))
> 
> -salam/salam/salam-

```python
len(a)

a.lower()

a.replace(old, new [,max])

a.rfind(str, beg=0, end=len(a))

a.rindex(str, beg=0, end=len(a))

a.split(str=" ", num=a.count(str))

a.splitlines( num=a.count('\n'))

a.startswith(str, beg=0, end=len(a))

a.endswith(str, beg=0, end=len(a))

a.swapcase()

a.isdeciaml()

max(str)

min(str)
```
> ascii code


In [32]:
a = 'salam'
print(a.islower())

True


## Lists

### Basic List Operations

```python
len([1, 2, 3])

[1, 2, 3] + [4, 5, 6]

['hi']*4

3 in [1, 2, 3]

for x in [1, 2, 3]: print(x)

```

### Indexing, Slicing

```python
l[2]

l[-2]

l[:]

l[1:]

l[:4]
```

### Built-in List Functions & Methods:

```python
max(l)

min(l)

list((1, 2, 3))
```

```python
l.append(m)

l.count(m)

l.extend((1,2,3))

l.index(m)

l.insert(index, m)

l.pop(obj=l[-1])

l.remove(m)

l.reverse()

l.sort([func])
```


## 2D list

> ?

###### get list with one input ?